Tasks/Common/SonarQubeHelpers/SonarQubeHelper.ps1

```diff
@@ -12,22 +12,30 @@ function EscapeArg
 
 
 # Set a variable in a property bag that is accessible by all steps
-# To retrieve the variable use $val = Get-Variable $distributedTaskContext "varName"
+# To retrieve the variable use GetTaskContextVariable
+# The variable will be available in both the current and the subsequent tasks
 function SetTaskContextVariable
 {
     param([string][ValidateNotNullOrEmpty()]$varName, 
           [string]$varValue)
         
-    #[Environment]::SetEnvironmentVariable($varName, $varValue)
-    Write-Host "##vso[task.setvariable variable=$varName;]$varValue"
+    [Environment]::SetEnvironmentVariable($varName, $varValue)
+    Set-VstsTaskVariable $varName $varValue
 }
 
 function GetTaskContextVariable()
 {
 	param([string][ValidateNotNullOrEmpty()]$varName)
         
-    #return ([Environment]::GetEnvironmentVariable($varName))
-	return Get-TaskVariable -Context $distributedTaskContext -Name $varName    
+    $value = [Environment]::GetEnvironmentVariable($varName);
+
+    if ([String]::IsNullOrWhiteSpace($value))
+    {
```

Comment:
The coreclr agent uses several processes to do its work, i.e. each build task is ran in a separate process. Variables needs to be pushed in each worker process, which the agent takes care of. However, the variable is not available in the current build task. So this is a workaround against taht.